specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,186 @@
+                {
+                    if (SUCCEEDED(result))
+                    {
+                        AsyncMessageBox(L"Completed", L"Clear Custom Partition Data");
+                    }
+                    else
+                    {
+                        std::wstringstream message;
+                        message << L"Failed: " << std::to_wstring(result) << L"(0x" << std::hex
+                                << result << L")" << std::endl;
+                        AsyncMessageBox(message.str(), L"Clear Custom Partition Data");
+                    }
+                    return S_OK;
+                }).Get()));
+          return S_OK;
+        }
+        
+```
+
+## .NET/WinRT
+```c#
+        // Sets custom storage partition identified by the partitionId, which uniquely
+        // identifies an application context.
+        void SetPartitionId(string partitionId)
+        {
+           m_webview.CustomStoragePartitionId = partitionId;
+        }
+        
+        // Clears all data in custom storage partition identified by the partitionId.
+        // Called when the application context is removed.
+        async void ClearPartitionData(string partitionId)
+        {
+          await m_webview.Profile.ClearCustomStoragePartitionDataAsync(partitionId);
+          MessageBox.Show(this, "Completed", "Clear Custom Partition Data");
+        }
+
+```
+
+# API Details
+## Win32 C++
+```
+interface ICoreWebView2_18 : IUnknown {
+  /// Gets the `CustomStoragePartitionId` property.
+  [propget] HRESULT CustomStoragePartitionId([out, retval] LPWSTR* customStoragePartitionId);
+
+  /// Sets the `CustomStoragePartitionId` property.
+  /// This feature requires enabling 2 experimental browser features to work properly.
+  /// These features will be enabled by default in the future.
+  /// Before these features are enabled by default, please enable them by ensuring
+  /// `--enable-features=ThirdPartyStoragePartitioning,PartitionedCookies` is set in
+  /// `AdditionalBrowserArguments` in `CoreWebView2EnvironmentOptions` used to create
+  /// CoreWebView2Environment. If these features are not enabled, all data are treated
+  /// as unpartitioned and stored in the global default location for the profile.
+  /// When it is set, the page in the WebView will act as if the page were hosted in a
+  /// top level site uniquely associated with the `partitionId` and have a separate 
+  /// storage partition as described in https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/
+  /// and separete cookie partition as described in https://developer.chrome.com/docs/privacy-sandbox/chips/
+  /// with all cookies partitioned.
+  /// If `customStoragePartitionId` is nullptr or empty string, the
+  /// `CustomStoragePartitionId` will be reset and the page inside the WebView
+  /// will work normally with data treated as unpartitioned.
+  /// The `customStoragePartitionId` parameter is case sensitive. The default is
```

Comment:
Is there any other info to give about the partition ID? Is there a length limit? Can it really just be any sort of string at all? If it is totally arbitrary and totally up to the end developer then it would be good to have the sample code show at least one partition ID value. Also would be good to state in the ref docs that the contents of the string are up to the end developer and the only condition is that it be unique within the user data folder.

Reply:
It is really be any string. We use hash internally. Will add "There is no restriction on the lenght or what characters
 can be used in partition id."